example/http/client/async-ssl/http_client_async_ssl.cpp

```diff
@@ -216,9 +228,12 @@ int main(int argc, char** argv)
     auto const target = argv[3];
     int version = argc == 5 && !std::strcmp("1.0", argv[4]) ? 10 : 11;
 
-    // The io_context is required for all I/O
+    // The io_context is one possible executor context
```

Comment:
You can't just change something like this in one example, it has to be changed in every example, and we need to agree upon the wording. I agree that it is a bit out of date with Chris' changes to I/O objects.

Reply:
The way I would structure this by choice is something like this:
```
asio::io_context ioc;

...

start(ioc.get_executor(), ...);

ioc.run();
```

This ends the dependency on the io_context type as early as possible in the program.

Reply:
How is 
```
asio::io_context ioc;
...
start(ioc.get_executor(), ...);
ioc.run();
```
any different from
```
    net::io_context ioc;
    auto exec = ioc.get_executor();
    std::make_shared<session>(exec, ctx)->run(host, port, target, version);
```
except that it adds one unnecessary function? `ioc` is still referenced the same number of times (twice, one for launching the client and the other for calling `run`).

Reply:
I guess it's a matter of personal style. It's no big deal.

Reply:
For me, the application's initialisation (starting initiatl operations on the executor) is a separate concern to the pumping of messages through the io_context.

I generally prefer to separate them completely.